src/main/java/sg/ncl/MainController.java

```diff
@@ -5016,4 +5020,184 @@ private TeamQuota extractTeamQuotaInfo(String responseBody) {
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+                switch (exceptionState) {
+                    case INVALID_EMAIL_ADDRESS_EXCEPTION:
+                        log.warn("Adding members by emails: Invalid email address");
+                        redirectAttributes.addFlashAttribute(MESSAGE, "Email address is not valid");
+                        break;
+                    default:
+                        log.warn("Adding members by emails : sio or deterlab adapter connection error");
+                        // possible sio or adapter connection fail
+                        redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+                        break;
+                }
+                return "redirect:add_member";
+            } catch (IOException e) {
+                throw new WebServiceRuntimeException(e.getMessage());
+            }
+        }
+        log.info("Adding members by email for team {} successful", teamId);
+        redirectAttributes.addFlashAttribute(MESSAGE_SUCCESS, "New members have been added successful!");
+        return "redirect:add_member";
+    }
+
+
+    // when user clicks password reset link in the email
+    @GetMapping(path = "/newMemberResetPassword", params = {"uid", "key"})
+    public String resetPasswordForNewMember(@NotNull @RequestParam("uid") final String uid,
+                                    @NotNull @RequestParam("key") final String key,
+                                              Model model) {
+        NewClassMemberPasswordResetForm newClassMemberPasswordResetForm = new NewClassMemberPasswordResetForm();
+        newClassMemberPasswordResetForm.setKey(key);
+        newClassMemberPasswordResetForm.setUid(uid);
+
+        model.addAttribute("newClassMemberPasswordResetForm", newClassMemberPasswordResetForm );
+        // redirect to the page for user to enter new password
+        return "new_member_reset_password";
+    }
+
+    // send to SIO to process resetting password for new member
+    @PostMapping("/new_member_reset_password")
+    public String processResetPasswordForNewMember(
+                        @ModelAttribute("newClassMemberPasswordResetForm") NewClassMemberPasswordResetForm newClassMemberPasswordResetForm
+                        ) throws WebServiceRuntimeException {
+
+
+        JSONObject obj = new JSONObject();
+        obj.put("firstName", newClassMemberPasswordResetForm.getFirstName());
+        obj.put("lastName", newClassMemberPasswordResetForm.getLastName());
+        obj.put("phone", newClassMemberPasswordResetForm.getPhone());
+        obj.put("key", newClassMemberPasswordResetForm.getKey());
+        obj.put("newPassword", newClassMemberPasswordResetForm.getPassword1());
+
+        String uid = newClassMemberPasswordResetForm.getUid();
+        String key = newClassMemberPasswordResetForm.getKey();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "key". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "key" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)